### ISSUE TITLE
config.yml - support qcow2 as PLATFORM value

### ISSUE THREAD

**Issue by amolari** [f5devcentral/f5-bigip-image-generator#21]
**Is your feature request related to a problem? Please describe.**
I need to create a local qcow2 image for later use on Alibaba.
The creation of such image works only if all information is provided on the command line, such as:
./build-image -i /opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso -e /opt/f5-isos/Hotfix-BIGIP-14.1.3.1.0.3.8-ENG.iso -p qcow2 -m all -b 2

Issue: I have to include files/packages (UPDATE_IMAGE_FILES) along with the image creation, so providing the image/enghf (with -i and -e cmdline parameters) is not sufficient. Using a config.yml file doesn't work I've tried to set PLATFORM: "qcow2" but I get then the error
`ValueError: Unexpected cloud type: qcow2
`
Please allow tthe build of a qcow2 using exclusively the config.yml file.

**Describe the solution you'd like**
Support qcow2 as PLATFORM value in config.yml
That will generate a local qcow2 image.

**Additional context**

**Comment by kmaloney** [comment]
I have tried to repro this, but I am guessing that I am missing something.  One question that I have is what version of Image Generator are you using?  I am trying to repro this on v1.10.  

I was able to put all of the vars that you provided in a config file (command: "./build-image -c config.yml").  I tried a few variations with and without quotes.  Are there other vars that you are using?

PLATFORM: qcow2
MODULES: all
BOOT_LOCATIONS: 2
ISO: /home/foo/images/BIGIP-14.1.3.1-0.0.8.iso
EHF_ISO: /home/foo/images/Hotfix-BIGIP-14.1.3.1.0.3.8-ENG.iso

UPDATE_IMAGE_FILES:
- source: "/home/foo/test_file"
  destination: "/config/startup"

**Comment by amolari** [comment]
My config:

```
REUSE: yes
BOOT_LOCATIONS: 2
MODULES: all
PLATFORM: qcow2
DISABLE_TELEMETRY: yes

CLOUD_IMAGE_NAME: "custom-BIGIP-14.1.3.1-0.3.8-BYOL-All-DO"
ISO: "/opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso"
EHF_ISO: "/opt/f5-isos/Hotfix-BIGIP-14.1.3.1.0.3.8-ENG.iso"

IMAGE_DIR: "/opt/ali_images"
LOG_FILE: "/opt/ali_images/generator.log"
UPDATE_IMAGE_FILES:
    -  source: "/home/demo/ric-custom/f5-declarative-onboarding-1.18.0-4.noarch.rpm"
       destination: "/var/config/rest/downloads/f5-declarative-onboarding-1.18.0-4.noarch.rpm"
```

I launch it with `./build-image -c config.yml`   (I use the latest version 1.10)

and I get the following error:

```
Generator info version matches current version 1.10
Starting the image build process
Extracting '/BIGIP14131/VERSION' from '/opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso'
Successfully extracted '/home/demo/f5-bigip-image-generator-1.10/src/lib/bash/../../../artifacts/BIGIP-14.1.3.1-0.0.8/qcow2/all_2slot/VERSION' from '/opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso'.
PRODUCT_NAME: BIGIP
PRODUCT_BASE_BUILD: 0.0.8
PRODUCT_BUILD: 0.0.8
PRODUCT_VERSION: 14.1.3.1
Extracting '/VERSION.LTM' from '/opt/f5-isos/Hotfix-BIGIP-14.1.3.1.0.3.8-ENG.iso'
Successfully extracted '/home/demo/f5-bigip-image-generator-1.10/src/lib/bash/../../../artifacts/BIGIP-14.1.3.1-0.0.8/qcow2/all_2slot/VERSION.LTM' from '/opt/f5-isos/Hotfix-BIGIP-14.1.3.1.0.3.8-ENG.iso'.
PRODUCT_NAME: BIGIP
PRODUCT_BASE_BUILD: 0.0.8
PRODUCT_BUILD: 0.3.8
PRODUCT_VERSION: 14.1.3.1
Extracting '/BIGIP14131/install/ve.info.json' from '/opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso'
Successfully extracted '/home/demo/f5-bigip-image-generator-1.10/src/lib/bash/../../../artifacts/BIGIP-14.1.3.1-0.0.8/qcow2/all_2slot/ve.info.json' from '/opt/f5-isos/BIGIP-14.1.3.1-0.0.8.iso'.
Traceback (most recent call last):
  File "/home/demo/f5-bigip-image-generator-1.10/src/bin/prepare_image.py", line 78, in <module>
    main()
  File "/home/demo/f5-bigip-image-generator-1.10/src/bin/prepare_image.py", line 55, in main
    ImageController.check_valid_name(args.platform, args.user_image_name)
  File "/home/demo/f5-bigip-image-generator-1.10/src/lib/python/image/image_controller.py", line 120, in check_valid_name
    cloud_image_name = ImageController.image_name_factory(cloud_type)
  File "/home/demo/f5-bigip-image-generator-1.10/src/lib/python/image/image_controller.py", line 115, in image_name_factory
    raise ValueError('Unexpected cloud type: {}'.format(cloud_type))
ValueError: Unexpected cloud type: qcow2
Outputting json file has not been requested.
```

**Comment by kmaloney** [comment]
Thanks for the detailed config.  The issue is with CLOUD_IMAGE_NAME.  Each cloud has their own rules for image naming, so the image generator validates the user-provided name near the beginning of the run to avoid failures later in the run.  We should be ignoring CLOUD_IMAGE_NAME for non-cloud platforms.  This issue will be tracked internally with VECLOUD-1146.

Prior to a fix, a workaround is to comment out the line with CLOUD_IMAGE_NAME.

**Comment by kmaloney** [comment]
Resolved in v1.11. CLOUD_IMAGE_NAME is now ignored for non-cloud platforms.
